src/solvers/refinement/string_refinement.h

```diff
@@ -164,39 +160,61 @@ class string_refinementt: public bv_refinementt
 
 exprt substitute_array_lists(exprt expr, size_t string_max_length);
 
+exprt concretize_arrays_in_expression(
+  const exprt &expr, std::size_t string_max_length);
+
 /// Utility function for concretization of strings. Copies concretized values to
 /// the left to initialize the unconcretized indices of concrete_array.
 /// \param concrete_array: the vector to populate
-/// \param initialized: the vector containing the indices of the concretized
-///   values
-/// \param last_concretized: initial value of the last concretized index
-template <typename T1, typename T2>
-void string_refinementt::pad_vector(
-  std::vector<T1> &concrete_array,
-  std::set<T2> &initialized,
-  T1 last_concretized) const
+/// \param initialized: set containing the indices of already concrete values
+template <typename T>
+void fill_in_vector(
```

Comment:
Instead of repeated `.find` calls here, use an explicit iterator and `*std::prev(it)` to get the previous initialized index.